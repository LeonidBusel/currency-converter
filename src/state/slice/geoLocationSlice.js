import { createSlice } from '@reduxjs/toolkit'

export const initialState = {
    isFetching: false,
    isLoaded: false,
    currency: null
};

export const geoLocationSlice = createSlice({
    name: 'geoLocation',
    initialState,
    reducers: {
        geoLocationFetch: state => {
            state.isFetching = true;
        },
        geoLocationFetchSuccess: (state, { payload }) => {
            state.isFetching = false;
            state.isLoaded = true;
            state.currency = payload;
        },
        geoLocationFetchFail: (state) => {
            state.isFetching = false;
            state.isLoaded = false;
            state.currency = null;
        }
    }
})

export const { geoLocationFetch, geoLocationFetchSuccess, geoLocationFetchFail } = geoLocationSlice.actions

export default geoLocationSlice.reducer