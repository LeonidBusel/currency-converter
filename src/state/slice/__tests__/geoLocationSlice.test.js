import geoLocationSlice, { initialState, geoLocationFetch, geoLocationFetchSuccess, geoLocationFetchFail } from "../geoLocationSlice";

const modifyState = {
    "currency": 55,
    "isLoaded": false,
    "isFetching": false
}

describe('geoLocationSlice', () => {
    it('handle geoLocationFetch correctly', () => {
        const actual = geoLocationSlice(initialState, geoLocationFetch());

        expect(actual).toMatchSnapshot();
    });

    it('handle geoLocationFetchSuccess correctly', () => {
        const actual = geoLocationSlice(initialState, geoLocationFetchSuccess(55));

        expect(actual).toMatchSnapshot();
    });

    it('handle geoLocationFetchFail correctly', () => {
        const actual = geoLocationSlice(modifyState, geoLocationFetchFail());

        expect(actual).toMatchSnapshot();
    });
});