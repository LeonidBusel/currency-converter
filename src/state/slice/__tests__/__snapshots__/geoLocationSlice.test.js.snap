// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`geoLocationSlice handle geoLocationFetch correctly 1`] = `
Object {
  "currency": null,
  "isFetching": true,
  "isLoaded": false,
}
`;

exports[`geoLocationSlice handle geoLocationFetchFail correctly 1`] = `
Object {
  "currency": null,
  "isFetching": false,
  "isLoaded": false,
}
`;

exports[`geoLocationSlice handle geoLocationFetchSuccess correctly 1`] = `
Object {
  "currency": 55,
  "isFetching": false,
  "isLoaded": true,
}
`;
